katdal/h5datav2_5.py

```diff
@@ -0,0 +1,631 @@
+from .lazy_indexer import LazyIndexer, LazyTransform
+
+logger = logging.getLogger(__name__)
+
+# Simplify the scan activities to derive the basic state of the antenna (slewing, scanning, tracking, stopped)
+SIMPLIFY_STATE = {'scan_ready': 'slew', 'scan': 'scan', 'scan_complete': 'scan', 'track': 'track', 'slew': 'slew'}
+
+SENSOR_PROPS = dict(DEFAULT_SENSOR_PROPS)
+SENSOR_PROPS.update({
+    '*activity': {'greedy_values': ('slew', 'stop'), 'initial_value': 'slew',
+                  'transform': lambda act: SIMPLIFY_STATE.get(act, 'stop')},
+    '*target': {'initial_value': '', 'transform': _robust_target},
+    # These float sensors are actually categorical by nature as they represent user settings
+    'RFE/center-frequency-hz': {'categorical': True},
+    'RFE/rfe7.lo1.frequency': {'categorical': True},
+    '*attenuation': {'categorical': True},
+    '*attenuator.horizontal': {'categorical': True},
+    '*attenuator.vertical': {'categorical': True},
+    'RFE/rfe.band.select.LCP': {'categorical': True},
+    'RFE/rfe.band.select.RCP': {'categorical': True},
+    'RFE/rfe.lo-intermediate.5GHz.frequency': {'categorical': True},
+    'RFE/rfe.lo-intermediate.6_7GHz.frequency': {'categorical': True},
+    'RFE/rfe.lo-final.frequency': {'categorical': True}
+})
+
+SENSOR_ALIASES = {
+    'nd_roach': 'roach.noise.diode.on',
+}
+
+def _calc_azel(cache, name, ant="ant1"):
+    """Calculate virtual (az, el) sensors from actual ones in sensor cache."""
+    real_sensor = 'Antennas/%s/%s' % (ant, 'pos.actual-scan-azim' if name.endswith('az') else 'pos.actual-scan-elev')
+    cache[name] = sensor_data = katpoint.deg2rad(cache.get(real_sensor))
+    return sensor_data
+
+VIRTUAL_SENSORS = dict(DEFAULT_VIRTUAL_SENSORS)
+VIRTUAL_SENSORS.update({'Antennas/{ant}/az': _calc_azel, 'Antennas/{ant}/el': _calc_azel})
+
+FLAG_NAMES = ('reserved0', 'static', 'cam', 'reserved3', 'detected_rfi', 'predicted_rfi', 'reserved6', 'reserved7')
+FLAG_DESCRIPTIONS = ('reserved - bit 0', 'predefined static flag list', 'flag based on live CAM information',
+                     'reserved - bit 3', 'RFI detected in the online system', 'RFI predicted from space based pollutants',
+                     'reserved - bit 6', 'reserved - bit 7')
+
+#--------------------------------------------------------------------------------------------------
+#--- Utility functions
+#--------------------------------------------------------------------------------------------------
+
+
+def make_pmodel_string(param_list):
+    """Take the h5 dataset that stores the AVN pointing model and convert it to a
+       string in the format that katpoint expects.
+    """
+    if not isinstance(param_list, h5py.Dataset):
+        raise ValueError("Error! param_list isn't an HDF5 dataset.")
+    model_string = ""
+    for param in param_list:
+        model_string += str(param) + " "
+    model_string = model_string[:-1]  # To take off the space on the end.
```

Comment:
A more compact way would be

```Python
model_string = ' '.join(str(param) for param in param_list)
```

which pretty much removes the need for this function.

Reply:
Good suggestion, thanks. Fixed in commit c4255d.